### PR TITLE
add spring security

### PR DESCRIPTION
**Pull request by MarrMika**

### CODE REVIEW

File: src/main/java/com/ita/if103java/ims/security/UserDetailsServiceImpl.java [hunk 1]
```diff
@@ -12,15 +15,24 @@
...
-        return new UserDetailsImpl(Optional.ofNullable(userDao.findByEmail(username))
+        User user = null;
+        Optional.ofNullable(user = userDao.findByEmail(username))
             .orElseThrow(() -> {
                 throw new UsernameNotFoundException("Error during obtain user with this username");
-            }));
+            });
+        AccountType type = accountDao.findById(user.getId()).getType();
 
+        return new UserDetailsImpl(user, type);
```
Comment (rostIvan) [inline src/main/java/com/ita/if103java/ims/security/UserDetailsServiceImpl.java]:
```java
User user = Optional
            .ofNullable(userDao.findByEmail(username))
            .orElseThrow(() -> new UsernameNotFoundException("Error during obtain user with this username"));
AccountType type = accountDao.findById(user.getId()).
return new UserDetailsImpl(user, type);
```

Comment (rostIvan) [reply src/main/java/com/ita/if103java/ims/security/UserDetailsServiceImpl.java]: Maybe it's never null, I don't know
```java
public User findByEmail(String email) {
        try {
            return jdbcTemplate.queryForObject(Queries.SQL_SELECT_USER_BY_EMAIL, userRowMapper, email);
        } catch (EmptyResultDataAccessException e) {
            throw new UserNotFoundException("Failed to obtain user during `select` {email = " + email + "}", e);
        }
```

Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/security/UserDetailsServiceImpl.java]: Yes, you are right. It makes sense

File: src/main/java/com/ita/if103java/ims/config/SecurityConfig.java [hunk 2]
```diff
@@ -21,13 +41,36 @@ protected void configure(HttpSecurity http) throws Exception {
...
     @Override
     public void configure(WebSecurity web) throws Exception {
-        web.ignoring().antMatchers("/v2/api-docs", "/configuration/ui", "/swagger-resources", "/configuration/security", "/swagger-ui.html", "/webjars/**");
+        web.ignoring()
+            .antMatchers("/v2/api-docs",
+                "/configuration/ui",
+                "/swagger-resources",
+                "/configuration/security",
+                "/swagger-ui.html",
+                "/webjars/**");
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/config/SecurityConfig.java]: I think we can combine these two lists of exclusion into one.
Could you please try to move "swagger" exclusion into "signin" exclusion list?

File: src/main/java/com/ita/if103java/ims/controller/UserController.java [hunk 3]
```diff
@@ -102,6 +104,7 @@ public void updatePassword(@AuthenticationPrincipal User user,
 
     @GetMapping("/me")
     public UserDto getCurrentUser(@AuthenticationPrincipal User user) {
+        System.out.println(user);
         return mapper.toDto(user);
     }
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/controller/UserController.java]: Is this method for test purposes?
@AuthenticationPrincipal User user
Shouldn't here be UserDetailsImpl class?

File: src/main/java/com/ita/if103java/ims/security/UserDetailsServiceImpl.java [hunk 4]
```diff
@@ -1,26 +1,31 @@
...
 
     @Override
-    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-        return new UserDetailsImpl(Optional.ofNullable(userDao.findByEmail(username))
-            .orElseThrow(() -> {
-                throw new UsernameNotFoundException("Error during obtain user with this username");
-            }));
+    public UserDetailsImpl loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userDao.findByEmail(username);
+        AccountType type = accountTypeDao.findById(user.getAccountId());
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/security/UserDetailsServiceImpl.java]: I think here is a mistake. First of all we need to find Account by account ID from User and then find AccountType by accountTypeId from Account.
@DashaBoichuk Please check and cooperate with Maria on this.

Comment (DashaBoichuk) [reply src/main/java/com/ita/if103java/ims/security/UserDetailsServiceImpl.java]: @MarrMika AccountType type = accountTypeDao.findById(accountDao.findById(user.getAccountId()).getTypeId());

File: src/main/java/com/ita/if103java/ims/config/SecurityConfig.java (hunk 2 above)
Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/config/SecurityConfig.java]: Of course)

File: src/main/java/com/ita/if103java/ims/security/UserDetailsServiceImpl.java (hunk 4 above)
Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/security/UserDetailsServiceImpl.java]: thanks)

File: src/main/java/com/ita/if103java/ims/controller/UserController.java (hunk 3 above)
Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/controller/UserController.java]: Yes. It is for test purposes. For now, I am thinking about enhancing it. Just use @AuthenticationPrincipal UserDetailsIml and return User or return UserDto

**Pull request merged**